### PR TITLE
adds new editions: reseller and managed hosting

### PR DESCRIPTION
**Pull request by biyeun**
http://manage.dimagi.com/default.asp?224069

screen shot 2016-04-20 at 6 27 41 pm
screen shot 2016-04-20 at 6 27 10 pm

@orangejenny @esoergel 
cc @proteusvacuum

### CODE REVIEW

File: corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+                    for product_rate in product_rates:
+                        product_rate.save()
+                        if hasattr(SoftwarePlanVersion, 'product_rates'):
+                            software_plan_version.product_rates.add(product_rate)
+                        elif hasattr(SoftwarePlanVersion, 'product_rate'):
+                            assert len(product_rates) == 1
+                            software_plan_version.product_rate = product_rate
+                        else:
+                            raise AccountingError('SoftwarePlanVersion does not have product_rate or product_rates field')
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py]: E501 line too long (122 > 115 characters)

File: corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+FEATURE_TYPES = [f[0] for f in FeatureType.CHOICES]
+PRODUCT_TYPES = [p[0] for p in SoftwareProductType.CHOICES]
+
+
+class Command(BaseCommand):
+    help = 'Bootstrap new plan editions effective April 2016: ' \
+           'Managed Hosting and Reseller'
+
+    option_list = BaseCommand.option_list + (
+        make_option('--dry-run', action='store_true',  default=False,
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py]: E241 multiple spaces after ','

File: corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                    help='Do not actually modify the database, just verbosely log what happen'),
+        make_option('--verbose', action='store_true',  default=False,
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py]: E241 multiple spaces after ','

File: corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                    help='Enable debug output'),
+        make_option('--testing', action='store_true',  default=False,
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py]: E241 multiple spaces after ','

File: corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
+\# Use modern Python
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_accounting_bootstrap_new_editions_042016.py]: E261 at least two spaces before inline comment
E901 SyntaxError: unexpected character after line continuation character

**Comment by esoergel** [comment]
Build fail looks real:
```
CommandError: Conflicting migrations detected (0027_bootstrap_new_editions, 0026_subscriber_domain_unique in accounting).
To fix them run 'python manage.py makemigrations --merge'
```

**Comment by biyeun** [comment]
yeah, looks like nick also got a migration in for accounting shortly before i had this up

**Comment by biyeun** [comment]
@esoergel ✅

**Comment by biyeun** [comment]
cc @orangejenny @esoergel ✅

**Pull request merged**